### PR TITLE
Revert S3 Support

### PR DESCRIPTION
**Pull request by akrherz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
*  Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Reverts S3 support added in #180 in hopes of helping #182 until a permanent fix can be found.

### CODE REVIEW

**Comment by akrherz** [comment]
@conda-forge-admin please rerender

**Review by xylar — APPROVED** [review]
@akrherz, thanks for doing this.  I think we need to revert this for now so we can make headway with migrations more broadly on conda-forge.  Obviously, we can revisit this as soon as there's a fix, as folks are exploring in #182

File: recipe/build.sh
Comment (ocefpaf) [inline recipe/build.sh:85]: My guess is that we can revert it only on the Windows builds, no?

Comment (akrherz) [reply recipe/build.sh:85]: I have a suspicion that something is wrong with the Linux builds as well since my project's CI started to fail with a strange segfault once the latest netcdf build landed.  I am not clever enough to debug GHA and I can't reproduce locally, so I could be wildly wrong.

Comment (xylar) [reply recipe/build.sh:85]: I also think it's a lot less confusing for users when all platforms have the same features whenever possible.  I realize that's not a super strong argument, though, so I could go either way.

@akrherz, it seems like random segfaults are disconcerting but a little hard to tie definitively to this change, so I would be hesitant to revert these changes on those grounds.  But maybe if we revert them for consistency and you find out that the segfaults do go away, that will lend more credence to the idea that they are related.

Comment (akrherz) [reply recipe/build.sh:85]: My again likely wrong suspicion is that the changes proposed here Unidata/netcdf-c#2741 are plaguing all platforms and not just windows.

Comment (xylar) [reply recipe/build.sh:85]: Okay, well it seems safest to revert for all platforms and then bring back the feature for all platforms in my view.

Comment (ocefpaf) [reply recipe/build.sh:85]: Yep. Just ignore my suggestion :grimacing: 

Definitely better to revert all and re-visit this in the near future. Sorry for the distraction and feel free to merge this when you think it is ready.

**Comment by akrherz** [comment]
Anybody willing to smash the green button here, so we can get migrations moving again?

**Comment by akrherz** [comment]
Again, a total FWIW and likely something I have screwed up, but I have found this updated build no longer does flakey things (segfaults, etc) on github actions with my own repos.  I'll be sure to watch when S3 support comes back in the future and if flakey things happen then too or not.

**Comment by xylar** [comment]
Sounds promising!  Let's hope when it comes back, it's more stable.

**Pull request merged**
